### PR TITLE
increase psalm level to 3

### PR DESCRIPTION
**Pull request by AlessandroMinoccheri**

### CODE REVIEW

File: src/Analyzer/FileVisitor.php [hunk 1]
```diff
@@ -28,12 +28,15 @@ public function enterNode(Node $node): void
...
-        if ($node instanceof Node\Expr\Instanceof_) {
+        if ($node instanceof Node\Expr\Instanceof_ && method_exists($node->class, 'toString')) {
             $this->classDescriptionBuilder
                 ->addDependency(new ClassDependency($node->class->toString(), $node->getLine()));
         }
 
-        if ($node instanceof Node\Expr\New_ && !($node->class instanceof Node\Expr\Variable)) {
+        if ($node instanceof Node\Expr\New_ &&
+            !($node->class instanceof Node\Expr\Variable) &&
+            method_exists($node->class, 'toString')
```
Comment (micheleorselli) [inline src/Analyzer/FileVisitor.php:38]: Should we extract a method for this particular condition? It find it a bit hard to get

Comment (AlessandroMinoccheri) [reply src/Analyzer/FileVisitor.php]: We can't move all conditions in another function for psalm reasons.
But I have improved a little bit.

File: src/Analyzer/FileVisitor.php [hunk 2]
```diff
@@ -63,4 +63,9 @@ public function leaveNode(Node $node): void
             $this->classDescriptions[] = $classDescription;
         }
     }
+
+    private function isNewNodeType(Node $node): bool
```
Comment (micheleorselli) [inline src/Analyzer/FileVisitor.php:67]: not sure i'm getting where this method is used. Apart from that, I think we should keep track why we are adding such checks and condition

Comment (AlessandroMinoccheri) [reply src/Analyzer/FileVisitor.php:67]: It was a refusal.

**Review by micheleorselli — APPROVED** [review]

**Pull request merged**